app/controllers/application_controller.rb

```diff
@@ -1,12 +1,40 @@
 class ApplicationController < ActionController::Base
-  # Prevent CSRF attacks by raising an exception.
-  # For APIs, you may want to use :null_session instead.
+  rescue_from ActiveRecord::RecordNotFound, with: :not_found
   protect_from_forgery with: :exception
+  before_action :set_locale
   before_filter :require_login
-  rescue_from ActiveRecord::RecordNotFound, with: :not_found
 
   private
 
+  def set_locale
+    if current_user
+      if current_user.locale
+        I18n.locale = current_user.locale
+      else
+        I18n.locale = http_accept_language.
+            compatible_language_from(I18n.available_locales) || I18n.default_locale
```

Comment:
Line is too long. [83/80]<br>Use 2 (not 4) spaces for indenting an expression spanning multiple lines.